test/extensions/filters/http/header_mutation/header_mutation_integration_test.cc

```diff
@@ -0,0 +1,166 @@
+#include "source/extensions/filters/http/header_mutation/config.h"
+
+#include "test/integration/http_integration.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace HeaderMutation {
+namespace {
+
+class HeaderMutationIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                                      public HttpIntegrationTest {
+public:
+  HeaderMutationIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()) {}
+
+  void initializeFilter() {
+    config_helper_.addFilter(R"EOF(
+name: donwstream-header-mutation
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.http.header_mutation.v3.HeaderMutation
+  mutations:
+    response_mutations:
+    - append:
+        header:
+          key: "downstream-global-flag-header"
+          value: "downstream-global-flag-header-value"
+        append_action: APPEND_IF_EXISTS_OR_ADD
+)EOF");
+
+    config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      envoy::extensions::upstreams::http::v3::HttpProtocolOptions http_protocol_options;
+      http_protocol_options.mutable_use_downstream_protocol_config();
+
+      // Upstream header mutation filter.
+      auto filter_0 = http_protocol_options.mutable_http_filters()->Add();
+      filter_0->set_name("upstream-header-mutation");
+      ProtoConfig header_mutation;
+      auto response_mutation =
+          header_mutation.mutable_mutations()->mutable_response_mutations()->Add();
+      response_mutation->mutable_append()->mutable_header()->set_key("upstream-global-flag-header");
+      response_mutation->mutable_append()->mutable_header()->set_value(
+          "upstream-global-flag-header-value");
+      response_mutation->mutable_append()->set_append_action(
```

Comment:
It's a bit jarring to have both YAML and proto generated methods used depending on downstream/upstream. If config helper is missing support for upstream filters, maybe now is the time to add it.

Reply:
ok, I will add new helper function for the upstream http filters.

Reply:
Seems that `prependFilter` is enough for this requirement.